stages/org.osbuild.chmod

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/python3
+"""
+Change file mode bits
+
+Change the file mode bits of one or more files or directories inside the tree.
+"""
+
+import subprocess
+import sys
+import os
+
+import osbuild.api
+
+SCHEMA = """
+"additionalProperties": false,
+"required": ["paths", "mode"],
+"properties": {
+  "paths": {
+    "type": "array",
+    "description": "Paths to operate on for changing mode bits",
+    "items": {
+      "type": "string"
+    }
+  },
+  "mode": {
+    "type": "string",
+    "description": "Symbolic or numeric octal mode"
+  },
+  "recursive": {
+    "type": "boolean"
+  }
+}
+
+"""
+
+def main(tree, options):
+    paths = options["paths"]
+    mode = options["mode"]
+    recursive = options.get("recursive", False)
+
+    paths = [os.path.join(tree, p.lstrip("/")) for p in paths]
+
+    arguments = [mode]
+    if recursive:
+        arguments += ["--recursive"]
+    arguments += paths
+
+    subprocess.run(["chmod", *arguments], check=True)
```

Comment:
After thinking about this, I am wondering, if we shouldn't use `os.chmod` instead. It is true that "historically" we would just use binaries which is convenient but leads to poor error reporting: one get a backtrace from the `subprocess.run` call and has to find the relevant message from the binary on stderr. It also makes it hard to pass errors back to `osbuild` in a more structured way (via something like `osbuild.api.error` -- something we don't yet have). The downside is that we have to do the recursive feature ourselves and probably would lose the symbolic variant (not sure this is something we need or want though).

Reply:
Supporting all the flexibility of `chmod(1)` gets out of hand fast.  I [wrote a parser](https://gist.github.com/achilleas-k/04ed3900efae0dce0590363e9bbc8628) that doesn't cover every case and it's already a bit much.  I think for this it's probably better to stick to shelling out.